cpp/coord/fixed_point.h

```diff
@@ -0,0 +1,286 @@
+// Copyright 2013-2015 the openage authors. See copying.md for legal info.
+
+#ifndef OPENAGE_COORD_FIXED_POINT_H_
+#define OPENAGE_COORD_FIXED_POINT_H_
+
+#include <cmath>
+#include <functional>
+
+#include "../util/misc.h"
+
+namespace openage {
+namespace coord {
+
+template<typename int_type, unsigned fractional_bits>
+struct FixedPoint {
+
+	int_type data;
+
+	static const int_type one = int_type(1) << fractional_bits;
+
+	static const int_type fraction_bitmask = one - 1;
+
+
+	// constructors
+	constexpr FixedPoint() : data(0) {
+	}
+
+	constexpr FixedPoint(int_type n) : data(n << fractional_bits) {
```

Comment:
We probably want those single-argument constructors to be `explicit` to prevent some implicit conversions.